pandas/_testing/asserters.py

```diff
@@ -399,9 +399,13 @@ def _get_ilevel_values(index, level):
     # skip exact index checking when `check_categorical` is False
     if check_exact and check_categorical:
         if not left.equals(right):
-            diff = (
-                np.sum((left._values != right._values).astype(int)) * 100.0 / len(left)
-            )
+            #  error: Value of type variable "_Number" of "sum" cannot be
+            # "Union[ExtensionArray, ndarray, Any]"
+            thesum = np.sum(
+                (left._values != right._values).astype(int)
```

Comment:
does adding an overload for `Type[str | float | int | complex | bool |object]` fix these false positives?

Reply:
or maybe move those into NpDtype from Dtype in pandas._typing.

Reply:
I changed `NpDtype`, which forced a bunch of changes elsewhere, mostly to get rid of `ignore`